core_v14/src/metadata/mod.rs

```diff
@@ -132,28 +148,20 @@ fn runtime_metadata_version(meta: &RuntimeMetadata) -> usize {
 #[derive(Debug)]
 struct MetadataPallet {
 	name: String,
-	calls: HashMap<u8, MetadataCall>,
+	/// Metadata may not contain call information. If it does,
+	/// it'll be here.
+	calls: Option<MetadataCalls>,
```

Comment:
Just guessing here, but maybe something like the `shell` parachain could have zero calls? (Well, hmm, that one has exactly one call right?). It'd be something pretty exotic, but unless it's illegal to have a pallet with zero calls we need something here that can represent such a thing.

Reply:
I wasn't sure either, but since the `frame-metadata` type allows for None I thought that instead of erroring immediately, I'd allow for the possibility and, if needed, handle errors closer to the calls that would be broken by it.